ehrapy/plot/_sa_plot.py

```diff
@@ -0,0 +1,218 @@
+    lines: list[tuple[ndarray | float, ndarray | float]] | None = None,
+    lines_color: list[str] | None | None = None,
+    lines_style: list[str] | None | None = None,
+    lines_label: list[str] | None | None = None,
+    xlim: tuple[float, float] | None = None,
+    ylim: tuple[float, float] | None = None,
+    **kwds,
+):
+    """Plots a Ordinary Least Squares (OLS) Model result, scatter plot, and line plot.
+
+    Args:
+        adata: :class:`~anndata.AnnData` object object containing all observations.
+        x: x coordinate, for scatter plotting
+        y: y coordinate, for scatter plotting
+        scatter_plot: If True, show scatter plot. Default is True
+        ols_results: List of RegressionResults. From ehrapy.tl.ols. Example: [result_1, result_2]
+        ols_color: List of colors for each ols_results. Example: ['red', 'blue']
+        xlabel: The x-axis label text
+        ylabel: The y-axis label text
+        figsize: Width, height in inches. Default is: [6.4, 4.8]
+        lines: List of Tuples of (slope, intercept) or (x, y). Plot lines by slope and intercept or data points. Example: plot two lines (y = x + 2 and y = 2*x + 1): [(1, 2), (2, 1)]
+        lines_color: List of colors for each line. Example: ['red', 'blue']
+        lines_style: List of line styles for each line. Example: ['-', '--']
+        lines_label: List of line labels for each line. Example: ['Line1', 'Line2']
+        xlim: Set the x-axis view limits. Required for only ploting lines using slope and intercept.
+        ylim: Set the y-axis view limits. Required for only ploting lines using slope and intercept.
+
+    Example:
+        .. code-block:: python
+
+            # Scatter plot and OLS regression plot
+            import ehrapy as ep
+            adata = ep.data.mimic_2(encoded=False)
+            co2_lm_result = ep.tl.ols(adata, var_names=['pco2_first', 'tco2_first'], formula='tco2_first ~ pco2_first', missing="drop").fit()
+            ep.pl.ols_plot(adata, x='pco2_first', y='tco2_first', ols_results=[co2_lm_result], ols_color=['red'], xlabel="PCO2", ylabel="TCO2")
+
+        .. image:: /_static/docstring_previews/ols_plot_1.png
+
+        .. code-block:: python
+
+            # Scatter plot and line plot
+            import ehrapy as ep
+            adata = ep.data.mimic_2(encoded=False)
+            ep.pl.ols_plot(adata, x='pco2_first', y='tco2_first', lines=[(0.25, 10), (0.3, 20)], lines_color=['red', 'blue'], lines_style=['-', ':'], lines_label=['Line1', 'Line2'])
+
+        .. image:: /_static/docstring_previews/ols_plot_2.png
+
+        .. code-block:: python
+
+            # Line plot only
+            import ehrapy as ep
+            ep.pl.ols_plot(lines=[(0.25, 10), (0.3, 20)], lines_color=['red', 'blue'], lines_style=['-', ':'], lines_label=['Line1', 'Line2'], xlim=(0, 150), ylim=(0, 50))
+
+        .. image:: /_static/docstring_previews/ols_plot_3.png
+    """
+    _, ax = plt.subplots(figsize=figsize)
```

Comment:
I would suggest that we use the figsize just like scanpy does it.

1. The parameter has None as a default value.
2. If it is None -> use the matplotlibrc settings which are the current ehrapy.settings.figsize settings
3. Can still be overwritten by passing it explicitly

Reply:
I checked scanpy and already changed this. 
At first I also implemented in the way you suggested, but I saw the default value in the matplotlib is (6.4, 4.8), so I changed to (6.4, 4.8).